src/sources/socket/mod.rs

```diff
@@ -1399,6 +1517,45 @@ mod test {
         assert_eq!(dgram, bytes);
     }
 
+    // TODO: maybe this should be in an integration test, such as `src/sources/redis/mod.rs` and `scripts/integration/redis`
+    // and so? There currently are no integration tests for the socket source
+    #[cfg(unix)]
+    #[tokio::test]
+    async fn unix_datagram_chunked_gelf_messages() {
+        assert_source_compliance(&SOCKET_PUSH_SOURCE_TAGS, async {
+            let (tx, rx) = SourceSender::new_test();
+            let in_path = tempfile::tempdir().unwrap().into_path().join("unix_test");
+            let mut config = UnixConfig::new(in_path.clone());
+            config.decoding = GelfDeserializerConfig::default().into();
+            let path = init_unix_with_config(tx, false, false, config).await;
+            let seed = 42;
+            let mut rng = SmallRng::seed_from_u64(seed);
+            // TODO: add a PR comment here stating that in MACOS at least, with bigger messages, the test fails
+            // with the `No buffer space available` error
+            let max_size = 20;
```

Comment:
(in MacOS at least) When testing with the same limits as the `udp_chunked_gelf_messages` of `max_size = 300` and `.repeat(1000)` it showed an OS error with code 55 and message `No buffer space available` (see [this issue](https://stackoverflow.com/questions/21973661/os-x-udp-send-error-55-no-buffer-space-available) for example).

What do you think?

Reply:
Huh, that is strange. The lower limits seem ok to me for testing.